Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -259,13 +279,176 @@ public class Klaviyo: NSObject  {
                 , "Property values must be of NSString, NSNumber, NSNull, NSDictionary, NSDate, or NSURL. Got: \(String(describing: properties[k as! NSCopying]))")
         }
     }
+}
+
+
+
+/// The main interface for the Klaviyo SDK.
+/// Create a new instance as follows:
+///
+/// ```swift
+/// let sdk = KlaviyoSDK()
+/// sdk.initialize(apiKey: "myapikey")
+/// ```
+///
+/// From there you can you can call the additional methods below to track events and profile.
+@_spi(KlaviyoPrivate)
+public struct KlaviyoSDK {
+    
+    /// Default initializer for the Klaviyo SDK.
+    @_spi(KlaviyoPrivate)
+    public init() {
+        
+    }
+    
+    private var state: KlaviyoState {
+        get {
+            environment.analytics.state()
+        }
+    }
     
-    private func dispatchOnMainThread(action: KlaviyoAction) {
-        Task {
-            await MainActor.run {
-                _ = environment.analytics.store.send(action)
-            }
+    /// Returns the email for the current user, if any.
+    @_spi(KlaviyoPrivate)
+    public var email: String? {
+        get {
+            state.email
+        }
+    }
+    
+    /// Returns the phoneNumber for the current user, if any.
+    @_spi(KlaviyoPrivate)
+    public var phoneNumber: String? {
+        get {
+            state.phoneNumber
+        }
+    }
+    
+    /// Returns the external id for the current user, if any.
+    @_spi(KlaviyoPrivate)
+    public var externalId: String? {
+        get {
+            state.externalId
         }
     }
     
+    @_spi(KlaviyoPrivate)
+    public var pushToken: String? {
+        get {
+            state.pushToken
+        }
+    }
+    
+    /// Initialize the swift SDK with the given api key.
+    /// - Parameter apiKey: your public api key from the Klaviyo console
+    /// - Returns: a KlaviyoSDK instance
+    @_spi(KlaviyoPrivate)
+    @discardableResult
+    public func initialize(with apiKey: String) -> KlaviyoSDK {
+        dispatchOnMainThread(action: .initialize(apiKey))
+        return self
+    }
+    
+    /// Set a profile in your Klaviyo account.
+    /// Future SDK calls will use this data when making api requests to Klaviyo.
+    /// NOTE: this will trigger a reset of existing
```

Comment:
This feels like a sentence fragment, "this will reset the existing currently tracked profile" maybe?

Reply:
also, can you link to another method, like @see resetProfile ?

Reply:
Not sure...

Reply:
Done.